.github/workflows/main.yml

```diff
@@ -165,8 +159,8 @@ jobs:
     - uses: actions/checkout@v2
     - uses: actions/setup-python@v2
     - name: set PY
-      run: echo "::set-env name=PY::$(python -c 'import hashlib, sys;print(hashlib.sha256(sys.version.encode()+sys.executable.encode()).hexdigest())')"
-    - uses: actions/cache@v1
+      run: echo "name=PY::$(python -c 'import hashlib, sys;print(hashlib.sha256(sys.version.encode()+sys.executable.encode()).hexdigest())')" >> $GITHUB_ENV
```

Comment:
can we use https://github.com/pre-commit/action  for that ? CC @asottile

Reply:
even better would be to move to https://pre-commit.ci -- https://results.pre-commit.ci is where that gets set up (~public alpha)